### ISSUE TITLE
[Bug] Crashes when typing with IME in block content editor.

### ISSUE THREAD

**Issue by naotone** [sanity-io/sanity#910]
I've found a bug when typing with IME(Input Method Editor). 
Actually, this is a problem with [slatejs](https://github.com/ianstormtaylor/slate), not sanity.
But I'll report it because this is a big problem for multi-byte language users like Japanese, Chinese and Korean.

Slate crashes when typing with IME after selected characters.
[https://github.com/ianstormtaylor/slate/issues/1861](https://github.com/ianstormtaylor/slate/issues/1861)

In addition, Sanity crashes when typing with IME at the beginning of a line.
(I typed nihongo (日本語) then hit enter.)
![2018-07-25 15 10 27](https://user-images.githubusercontent.com/957200/43222112-2510728e-901d-11e8-9151-d9cffc4f10ca.gif)

I've found a pull request to fix this problems at slate's repo.
[https://github.com/ianstormtaylor/slate/pull/1982](https://github.com/ianstormtaylor/slate/pull/1982)

Lots of bug with IME reported at slate's repo
[https://github.com/ianstormtaylor/slate/issues?utf8=✓&q=is%3Aissue+is%3Aopen+ime](https://github.com/ianstormtaylor/slate/issues?utf8=✓&q=is%3Aissue+is%3Aopen+ime)

Now I'm coding for my first project with sanity and nuxt.js. This is so fun and I can feel the future of CMS!

---
**Which versions of Sanity are you using (sanity versions)?**
0.132.5

**What operating system are you using?**
macOS High Sierra 10.13.6
Safari and Chrome
macOS's default IME(Japanese and Simplified Chinese) and [Google Japanese IME]()

**Which versions of Node.js / npm are you running?**
node v10.4.1 / 6.1.0

**What did you do?**
**What did you expect to see?**
**What did you see instead?**

**Comment by DongeunPaeng** [comment]
This issue remains regarding Korean.
After clicking the 'publish' button, the last character before the ending punctuation mark(.) disappears.

**Comment by DongeunPaeng** [comment]
The last update to 2.1.0 has worsened this issue.
CJK now can't even be typed straight, which means that my cursor goes to the very first place in the sentence often, suddenly while typing.

Please help! I hope someone would make improvements on this.

**Comment by naotone** [comment]
It should be fixed with despatch EditorChange events after browser return "compositionend" if IME is active.
I'll try it and come back here.

https://github.com/sanity-io/sanity/blob/00c0016aed650f57d5349347c404e333f34a6264/packages/%40sanity/form-builder/src/inputs/PortableText/PortableTextInput.tsx#L133-L161
https://developer.mozilla.org/en-US/docs/Web/API/Element/compositionend_event
https://www.w3.org/TR/uievents/#keys-IME

Hi @DongeunPaeng,
Version 1.149.x's editor is stable on Chrome! Latest version of 1.x should also works too.

**Comment by DongeunPaeng** [comment]
Thank you @naotone,
I'll try it myself. You're the best 👍

**Comment by DongeunPaeng** [comment]
Hi @naotone ,

Would you kindly teach me how to upgrade the slate to version 1.149.x?
I'm afraid I don't know how to upgrade a specific package (in this case, 'slate').

I've tried `yarn upgrade` and `yarn upgrade slate`, but neither couldn't upgrade the package's version from @0.58.4 to the desired version 1.149.x.

**Comment by naotone** [comment]
@DongeunPaeng 
Hi, try this!
```
"@sanity/base": "^0.147.8",
"@sanity/cli": "^0.147.8",
"@sanity/components": "^0.147.9",
"@sanity/core": "^0.147.9",
"@sanity/dashboard": "^0.147.8",
"@sanity/default-layout": "^0.147.9",
"@sanity/default-login": "^0.147",
"@sanity/desk-tool": "^0.147.9",
"@sanity/vision": "^0.147.9",
```

**Comment by naotone** [comment]
Hi @vicbergquist,

Could somebody in Sanity help us to solve this issue?
I couldn't find the way to add onCompositionStart and onCompositionEnd to Portable text component.
So I added global events for "compositionstart" and "compositionend" and save the state with react’s useState, and this works well.
I can help you with debugging and so on.

Issues
- [ ] Mutate editor change events after composition is end.
- [ ] Disable key inputs(especially arrow keys) while composition is active.
It must be 12345, not 12534
![](https://user-images.githubusercontent.com/957200/102838518-66156680-43cc-11eb-816b-a09a5f9e3511.gif)

**Comment by s7tya** [comment]
Hey! Any updates on this issue?
Look like the issue has already fixed on original slate-react repository.

**Merged PR of slate-react (original):** 
[ianstormtaylor/slate@`a297343` (#2218)](https://github.com/ianstormtaylor/slate/pull/2218/commits/a2973439bff634480f572fc32d0169eb3f76da68)

( Only on Chrome, cannot reproduce on Firefox. )

https://user-images.githubusercontent.com/53410646/143442387-9571e9a7-0aa9-4ce3-acc7-4907c0315ab1.mp4

**Comment by skogsmaskin** [comment]
> Hey! Any updates on this issue? Look like the issue has already fixed on original slate-react repository.

Yes, we are currently working on this.

**Comment by s7tya** [comment]
Alright! Thanks :)

**Comment by skogsmaskin** [comment]
Hey! We have released a new canary where hopefully this will be fixed. I need help in confirming this, so if you are able to test it out that would be great.

Install it with:
`sanity upgrade --tag=canary --save-exact` in the root of your studio folder.

**Comment by s7tya** [comment]
It works! I'd appreciate it.

**Comment by skogsmaskin** [comment]
Closing this issue as this is now released as part of version `2.25.0`
